### PR TITLE
v0.7 release prep

### PR DESCRIPTION
**Pull request by bobbinth**
This PR updates README files and user docs in preparation for v0.7.0 release of the VM.

As a part of this update I also re-ran all benchmarks and added benchmarks for recursive proofs. Overall, the performance has improved by about 30% as compared to the last time the benchmarks were updated.

### CODE REVIEW

File: miden/README.md [hunk 1]
```diff
@@ -16,32 +16,33 @@ All Miden programs can be reduced to a single 32-byte value, called program hash
...
 * `stack_inputs: StackInputs` - a set of public inputs with which to execute the program.
-* `advice_provider: AdviceProvider` - an instance of an advice provider that yields secret, non-deterministic inputs to the prover.
+* `host: Host` - an instance of a `Host` which can be used to supply non-deterministic inputs to the VM and receive messages from the VM.
+* `options: ExecutionOptions` - a set of options for executing the specified program (e.g., max allowed number of cycles).
 
-The `execute()` function returns a `Result<ExecutionTrace, ExecutionError>` which will contain the execution trace of the program if the execution was successful, or an error, if the execution failed. You can inspect the trace to get the final state of the VM out of it, but generally, this trace is intended to be used internally by the prover during proof generation process.
+The function returns a `Result<ExecutionTrace, ExecutionError>` which will contain the execution trace of the program if the execution was successful, or an error, if the execution failed. Internally, the VM then passes this execution trace to the prover to generate a proof of a correct execution of the program.
```
Comment (Al-Kindi-0) [inline miden/README.md:39]: `execute()` missing?

File: processor/README.md [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
 * `stack_inputs: StackInputs` - a set of public inputs with which to execute the program.
-* `advice_provider: AdviceProvider` - an instance of an advice provider that yields secret, non-deterministic inputs to the prover.
+* `host: Host` - an instance of a `Host` which can be used to supply non-deterministic inputs to the VM and receive messages from the VM.
+* `options: ExecutionOptions` - a set of options for executing the specified program (e.g., max allowed number of cycles).
 
-The `execute()` function returns a `Result<ExecutionTrace, ExecutionError>` which will contain the execution trace of the program if the execution was successful, or an error, if the execution failed. Internally, the VM then passes this execution trace to the prover to generate a proof of a correct execution of the program.
+The function returns a `Result<ExecutionTrace, ExecutionError>` which will contain the execution trace of the program if the execution was successful, or an error, if the execution failed. Internally, the VM then passes this execution trace to the prover to generate a proof of a correct execution of the program.
```
Comment (Al-Kindi-0) [inline processor/README.md:12]: same remark as above

**Review by Al-Kindi-0 — APPROVED** [review]
All looks good! Thank you!
Just a couple of minor remarks.

File: miden/README.md [hunk 3]
```diff
@@ -16,32 +16,33 @@ All Miden programs can be reduced to a single 32-byte value, called program hash
 Currently, there are 3 ways to get values onto the stack:
 
 1. You can use `push` instruction to push values onto the stack. These values become a part of the program itself, and, therefore, cannot be changed between program executions. You can think of them as constants.
-2. The stack can be initialized to some set of values at the beginning of the program. These inputs are public and must be shared with the verifier for them to verify a proof of the correct execution of a Miden program. The number of elements at the top of the stack which can receive an initial value is limited to 16.
+2. The stack can be initialized to some set of values at the beginning of the program. These inputs are public and must be shared with the verifier for them to verify a proof of the correct execution of a Miden program. While it is possible to initialize the stack with a large number of values, we recommend limiting the number of initial values to a few doze at most as each initial value beyond 16 increases verifier complexity.
```
[the remark this answers is not in the document]
Comment (bobbinth) [reply miden/README.md:19]: Fixed!

File: miden/README.md (hunk 1 above)
Comment (bobbinth) [reply miden/README.md:39]: I think in this context we are talking about the `execute()` function already - so, may be redundant to say it.

**Pull request merged**